apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.Data/ConnectionPoolOptions.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright 2017 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+
+namespace Google.Cloud.Spanner
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    public class ConnectionPoolOptions
+    {
+        /// <summary>
+        /// The default instance of connection pool options.
+        /// </summary>
+        public static readonly ConnectionPoolOptions Instance = new ConnectionPoolOptions();
+        private ConnectionPoolOptions() {
+            
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public static int GrpcChannelCount
+        {
+            get { throw new NotImplementedException(); }
+            set { throw new NotImplementedException(); }
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public int KeepAliveIntervalMinutes
+        {
+            get { throw new NotImplementedException(); }
+            set { throw new NotImplementedException(); }
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public int MaxIdleSessionPoolSize
+        {
+            get { throw new NotImplementedException(); }
+            set { throw new NotImplementedException(); }
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public int MaxSessionPoolSize
+        {
+            get { throw new NotImplementedException(); }
+            set { throw new NotImplementedException(); }
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public int MinSessionPoolSize
+        {
+            get { throw new NotImplementedException(); }
+            set { throw new NotImplementedException(); }
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public ResourcesExhaustedBehavior ResourcesExhaustedBehavior
+        {
+            get { throw new NotImplementedException(); }
+            set { throw new NotImplementedException(); }
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public static int Timeout
```

Comment:
It's the overall timeout for a command/request.  internally, we will have settings for backoff time, etc -- but won't expose that to start.